apps/shipments/views.py

```diff
@@ -196,3 +196,23 @@ def create(self, request, *args, **kwargs):
         permission_link = PermissionLink.objects.create(**serializer.validated_data)
 
         return Response(PermissionLinkSerializer(permission_link).data, status=status.HTTP_201_CREATED)
+
+
+class DeviceShipmentsHistoryListView(generics.ListAPIView):
+    permission_classes = ((permissions.IsAuthenticated, DeviceShipmentHistoryPermission) if settings.PROFILES_ENABLED
+                          else (permissions.AllowAny,))
+    resource_name = 'ShipmentHistory'
+    serializer_class = DeviceShipmentsHistorySerializer
+    filter_backends = (filters.SearchFilter, filters.OrderingFilter, DjangoFilterBackend,)
+    search_fields = ('shippers_reference', 'forwarders_reference', 'owner_id',)
+    ordering_fields = ('created_at', 'pickup_est', 'delivery_est', 'historical_date',)
+    filterset_class = ShipmentHistoryFilter
+    queryset = Shipment.history.all()
```

Comment:
Instead of returning list of copies of versions of the Shipment over time, I think it would be useful if this endpoint returned a list of diffs, so you can more easily see which fields changed over time.
https://django-simple-history.readthedocs.io/en/2.7.0/history_diffing.html
You can query Shipment.history.all(), but in the view you should loop over all the Shipment history objects (chronologically) and use the .diff_against() method to get the delta between them. And then this endpoint would serialize and return a list of diffs.

Reply:
Note that it should also contain diffs for the associated Locations